apps/src/blockly/addons/cdoFieldToggle.js

```diff
@@ -0,0 +1,123 @@
+import GoogleBlockly from 'blockly/core';
+
+export default class CdoFieldToggle extends GoogleBlockly.Field {
+  /**
+   * This is a customized field which the user clicks to toggle between two different states,
+   * for example, displaying or hiding the the block flyout.
+   *
+   * @param {Object} options - The options for constructing the class.
+   * @param {Function} options.onClick - The function that handles the field's editor.
+   * @param {SVGElement} options.icon1 - SVG <tspan> element - this is the icon that is initially displayed on the button.
```

Comment:
do the names `icon1` and `icon2` come from blockly? If not could we make them more specific?

Reply:
They do not from mainline! I can rename. Thanks!